aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/UpdateHandler.java

```diff
@@ -1,59 +1,85 @@
 package software.amazon.rds.dbcluster;
 
-import java.util.Collections;
-import java.util.List;
-import java.util.Optional;
+import static software.amazon.rds.dbcluster.ModelAdapter.setDefaults;
+import static software.amazon.rds.dbcluster.Translator.cloudwatchLogsExportConfiguration;
+
+import java.util.Collection;
+
 import software.amazon.awssdk.services.rds.RdsClient;
-import software.amazon.awssdk.services.rds.model.DbClusterRoleNotFoundException;
+import software.amazon.awssdk.services.rds.model.CloudwatchLogsExportConfiguration;
 import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
-import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
-import software.amazon.cloudformation.proxy.ProxyClient;
-import software.amazon.cloudformation.proxy.ProgressEvent;
 import software.amazon.cloudformation.proxy.Logger;
-
-import static software.amazon.rds.dbcluster.ModelAdapter.setDefaults;
-import static software.amazon.rds.dbcluster.Translator.cloudwatchLogsExportConfiguration;
-import static software.amazon.rds.dbcluster.Translator.removeRoleFromDbClusterRequest;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.rds.common.handler.Commons;
+import software.amazon.rds.common.handler.HandlerConfig;
+import software.amazon.rds.common.handler.Tagging;
 
 public class UpdateHandler extends BaseHandlerStd {
 
+    public UpdateHandler() {
+        this(HandlerConfig.builder().build());
+    }
+
+    public UpdateHandler(final HandlerConfig config) {
+        super(config);
+    }
+
     protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
-        final AmazonWebServicesClientProxy proxy,
-        final ResourceHandlerRequest<ResourceModel> request,
-        final CallbackContext callbackContext,
-        final ProxyClient<RdsClient> proxyClient,
-        final Logger logger) {
-
-      return ProgressEvent.progress(setDefaults(request.getDesiredResourceState()), callbackContext)
-          .then(progress -> modifyDBCluster(proxy, proxyClient, progress, cloudwatchLogsExportConfiguration(request)))
-          .then(progress -> waitForDBClusterAvailableStatus(proxy, proxyClient, progress))
-          .then(progress -> removeAssociatedRoles(proxy, proxyClient, progress, setDefaults(request.getPreviousResourceState()).getAssociatedRoles()))
-          .then(progress -> addAssociatedRoles(proxy, proxyClient, progress, progress.getResourceModel().getAssociatedRoles()))
-          .then(progress -> tagResource(proxy, proxyClient, progress))
-          .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
+            final AmazonWebServicesClientProxy proxy,
```

Comment:
this handler must be refactored. It must handle conditional properties and throw not updatable exception when needed. For instance EngineVersion is no longer create only property and must update gracefully when it's an upgrade and must throw not updatable exception when it's a downgrade so it will trigger a replacement.

Reply:
Added an explicit check for `GlobalClusterIdentifier` and `Engine`